components/playlist/playlist_download_request_manager.cc

```diff
@@ -115,10 +115,28 @@ void PlaylistDownloadRequestManager::RunMediaDetector(Request request) {
   callback_for_current_request_ = std::move(request.callback);
   DCHECK(!callback_for_current_request_.is_null());
 
-  GURL url(request.url);
-  DCHECK(url.is_valid());
-  web_contents_->GetController().LoadURLWithParams(
-      content::NavigationController::LoadURLParams(url));
+  if (absl::holds_alternative<std::string>(request.url_or_contents)) {
+    CreateWebContents();
+    GURL url(absl::get<std::string>(request.url_or_contents));
+    DCHECK(url.is_valid());
+    DCHECK(web_contents_);
+    web_contents_->GetController().LoadURLWithParams(
```

Comment:
does this get loaded in incognito/tor contexts for playlist requests associated with incognito/tor windows?

Reply:
@sangwoo108 i strongly feel we should make sure there is no way to enable playlist UI in incognito/tor windows. if i'm visiting a site in Tor, i probably never want anything associated with the site to be loaded without the Tor proxy. if the UI is enabled in Tor windows, all the requests for content associated with the Tor window should go through Tor.

Reply:
Thanks! I'll make sure that.